scanner/rails.go

```diff
@@ -328,3 +329,12 @@ func RailsCallback(appName string, srcInfo *SourceInfo, plan *plan.LaunchPlan) e
 
 	return nil
 }
+
+func RailsFailureCallback(msg string) string {
+	if !strings.Contains(msg, "https://") {
+		msg += "\n\nSee https://fly.io/docs/rails/getting-started/existing/#common-initial-deployment-issues\n" +
+			"for suggestions on how to resolve common deployment issues."
+	}
+
+	return msg
+}
```

Comment:
It might be good to return a specific error type for docs links so we can format them from the caller? e.g. 

```go
return &scanner.Error{
	DocsURL: "/docs/rails/getting-started/existing#common-initial-deployment-issues",
}
```

Reply:
I see a number of failures that are billing errors, and those errors contain a URL. In such cases it wouldn't be helpful to add a link to a language specific FAQ.

Reply:
I'm not clear how _format them from the caller?_ would work.  I (now) see code that formats a generic error, but I don't see how `cli.go`'s `printError` function would know how to handle a Rails specific error.  Can you elaborate on how this would work?

Reply:
Another common error appears to be `Validation failed: Name has already been taken`.  It might be better if a list of common errors is checked against _before_ any call to scanner specific failure callbacks.  That way the list can be in one place.

Reply:
Changed this to only add a suggestion if one is not already present.  This enabled code that creates errors to have their own (more specific) suggestions; and for the framework to add one if not present (or - at its option, modify or replace the suggestion).